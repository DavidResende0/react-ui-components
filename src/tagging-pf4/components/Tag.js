import * as React from 'react';
import PropTypes from 'prop-types';
import { Badge } from '@patternfly/react-core'

import TaggingPropTypes from '../TaggingPropTypes';
import BadgeWithDelete from './BadgeWithDelete'

const tooltip = text => <Tooltip id="tooltip">{text}</Tooltip>;

const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="pf4-tag">
    <BadgeWithDelete
      onDeleteClick={onTagDeleteClick}
      description={tagValue.description}
      truncate={truncate}
    />
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagValue: TaggingPropTypes.value,
  truncate: PropTypes.func.isRequired,
};

export default Tag;
