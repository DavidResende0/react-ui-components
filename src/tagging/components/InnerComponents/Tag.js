import * as React from 'react';
import PropTypes from 'prop-types';
import { Label, OverlayTrigger, Tooltip } from 'patternfly-react';
import TaggingPropTypes from '../TaggingPropTypes';

const tooltip = text => <Tooltip id="tooltip">{text}</Tooltip>;

const Tag = ({
  onTagDeleteClick, tagCategory, tagValue, truncate,
}) => (
  <li key={tagValue.id} className="tag">
    <OverlayTrigger placement="bottom" overlay={tooltip(tagValue.description)}>
      <Label
        key={tagValue.id}
        bsStyle="primary"
        onRemoveClick={() => onTagDeleteClick(tagCategory, tagValue)}
        className="tagColor"
      >
        {truncate(tagValue.description)}
      </Label>
    </OverlayTrigger>
  </li>
);

Tag.propTypes = {
  onTagDeleteClick: PropTypes.func.isRequired,
  tagCategory: TaggingPropTypes.category,
  tagValue: TaggingPropTypes.value,
  truncate: PropTypes.func.isRequired,
};

export default Tag;
