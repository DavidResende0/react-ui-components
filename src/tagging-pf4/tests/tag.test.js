import React from 'react';
import Tag from '../components/Tag';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const tagValue = { description: 'duck', id: 1 };
function onDelete(x) {
  return x;
}

describe('Tag Component', () => {
  it('match snapshot', () => {
    const component = shallow(
      <Tag
        tagValue={tagValue}
        onTagDeleteClick={onDelete}
        truncate={jest.fn()}
      />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
