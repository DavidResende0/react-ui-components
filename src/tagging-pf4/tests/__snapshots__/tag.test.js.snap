// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component match snapshot 1`] = `
<li
  className="pf4-tag"
  key="1"
>
  <BadgeWithDelete
    className=""
    description="duck"
    onDeleteClick={[Function]}
    truncate={[MockFunction]}
  />
</li>
`;
